station_performance_pipeline/extract.py

```diff
@@ -0,0 +1,89 @@
+from os import environ as ENV
+from datetime import datetime
+
+import requests
+from requests.auth import HTTPBasicAuth
+from dotenv import load_dotenv
+
+
+def train_station_arrival_data(station_crs: str) -> list[dict]:
+    '''This function accepts the crs code of a station and returns a list of dictionaries.
+    Each dictionary represents a train that arrived at the station on this day.
+    The dictionary has the keys related to the expected and actual times of arrival and departure.'''
+
+    now = datetime.now()
+    response = requests.get(
+        F"https://api.rtt.io/api/v1/json/search/{station_crs}/{now.strftime('%Y')}/{now.strftime('%m')}/{now.strftime('%d')}", auth=HTTPBasicAuth(ENV["REALTIME_API_USER"], ENV["REALTIME_API_PASS"]))
+
+    todays_station_data = response.json()
+
+    # Data on all of the instances of services running stopping at teh station
+    services_data = todays_station_data["services"]
+
+    arrived_services = []
+    for service in services_data:
+
+        if "cancelReasonCode" not in service["locationDetail"].keys():
+
+            arrived_service = {}
+            arrived_service["service_id"] = service["serviceUid"]
+            if "gbttBookedArrival" in service["locationDetail"].keys():
+                arrived_service["scheduled_arrival"] = service["locationDetail"]["gbttBookedArrival"]
+                arrived_service["actual_arrival"] = service["locationDetail"]["realtimeArrival"]
+            if "gbttBookedDeparture" in service["locationDetail"].keys():
+                arrived_service["scheduled_departure"] = service["locationDetail"]["gbttBookedDeparture"]
+                arrived_service["actual_departure"] = service["locationDetail"]["realtimeDeparture"]
+            # maybe don't need to search for this every time
+            arrived_service["station_crs"] = service["locationDetail"]["crs"]
```

Comment:
you noted this, but yeah, could just set it to equal the argument of the function i.e. 
```suggestion
            arrived_service["station_crs"] = station_crs
```

Similarly, outside the for loop, we could get the associated station name once, and then assign it from that variable i.e.
```suggestion
            station_name = service[0]["locationDetail"]["description"]
            
            for service in services_data:
                arrived_service["station_name"] = station_name
```

Reply:
Great, that's what I was looking to do!